packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -565,6 +565,34 @@ the configured path by running this command: flutter config --android-studio-dir
     }
   }
 
+  static Version? _parseVersion(String? text) {
+    Match? match;
+    // Example string fort EAP: EAP AI-242.21829.142.2422.12358220
+    // We try to capture 2422 here.
+    final RegExp intellijPattern = RegExp(r'EAP\s+[A-Z]{2}-\d+\.\d+\.\d+\.(\d+)\.\d+');
+    match = intellijPattern.firstMatch(text ?? '');
+
+    if (match == null) {
+      return Version.parse(text);
+    }
+
+    final String? matched = match.group(1);
+
+    // length of 4 is because that how version is encrypted: first two digits
+    // for year (part of major version), third for minor version, fourth for patch.
+    if (matched == null || matched.length != 4) {
+      return null;
+    }
+
+    final int? major = int.tryParse('20${matched[0]}${matched[1]}');
+    final int? minor = int.tryParse(matched[2]);
+    final int? patch = int.tryParse(matched[3]);
+    if (major == null || minor == null || patch == null) {
+      return null;
+    }
```

Comment:
This wouldn't work in any version where the minor or patch number is greater than 9. However, looking at the release history for Android Studio, this probably won't be a problem unless things change. LGTM.

Reply:
Agree, but introducing two-digits versioning will completely break this format, so I'm not sure that studio's team will allow that. But, from the other side, this is undocumented behaviour, so they are kinda free to do that :( But yeah, I do not think that it's possible and even justified here to take that possibility in account, because of code will become too cumbersome